integrations/access/datadog/client.go

```diff
@@ -0,0 +1,236 @@
+/*
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/integrations/access/common"
+	"github.com/gravitational/teleport/integrations/lib/logger"
+	pd "github.com/gravitational/teleport/integrations/lib/plugindata"
+)
+
+const (
+	datadogMaxConns    = 100
+	datadogHTTPTimeout = 10 * time.Second
+	statusEmitTimeout  = 10 * time.Second
+)
+
+const (
+	IncidentWritePermissions = "incident_write"
+)
```

Comment:
Can we please add a godoc explaining what this permission allows?
A short sentence and link to https://docs.datadoghq.com/account_management/rbac/permissions/#case-and-incident-management should be enough.